plugin/src/main/cpp/extensions/openxr_fb_environment_depth_extension_wrapper.cpp

```diff
@@ -0,0 +1,625 @@
+	GDEXTENSION_INIT_XR_FUNC_V(xrStopEnvironmentDepthProviderMETA);
+	GDEXTENSION_INIT_XR_FUNC_V(xrSetEnvironmentDepthHandRemovalMETA);
+	GDEXTENSION_INIT_XR_FUNC_V(xrCreateEnvironmentDepthSwapchainMETA);
+	GDEXTENSION_INIT_XR_FUNC_V(xrGetEnvironmentDepthSwapchainStateMETA);
+	GDEXTENSION_INIT_XR_FUNC_V(xrDestroyEnvironmentDepthSwapchainMETA);
+	GDEXTENSION_INIT_XR_FUNC_V(xrEnumerateEnvironmentDepthSwapchainImagesMETA);
+	GDEXTENSION_INIT_XR_FUNC_V(xrAcquireEnvironmentDepthImageMETA);
+
+	return true;
+}
+
+bool OpenXRFbEnvironmentDepthExtensionWrapper::prepare() {
+	// TODO: Verify that the shader parameters exist, if possible
+
+	if (!fb_environment_depth_ext) {
+		UtilityFunctions::print("fb_environment_depth extension not initialized");
+		return false;
+	}
+	if (provider != nullptr) {
+		return true;
+	}
+
+	XrEnvironmentDepthProviderCreateInfoMETA provider_create_info = {
+		XR_TYPE_ENVIRONMENT_DEPTH_PROVIDER_CREATE_INFO_META, // type
+		nullptr, // next
+		0, // createFlags
+	};
+
+	XrResult result;
+	result = xrCreateEnvironmentDepthProviderMETA(SESSION, &provider_create_info, &provider);
+	if (XR_FAILED(result)) {
+		UtilityFunctions::print("xrCreateEnvironmentDepthProviderMETA failed!");
+		UtilityFunctions::print(get_openxr_api()->get_error_string(result));
+		return false;
+	}
+
+	XrEnvironmentDepthSwapchainCreateInfoMETA swapchain_create_info = {
+		XR_TYPE_ENVIRONMENT_DEPTH_SWAPCHAIN_CREATE_INFO_META, // type
+		nullptr, // next
+		0, // createFlags
+	};
+
+	result = xrCreateEnvironmentDepthSwapchainMETA(provider, &swapchain_create_info, &swapchain);
+	if (XR_FAILED(result)) {
+		UtilityFunctions::print("xrCreateEnvironmentDepthSwapchainMETA failed!");
+		UtilityFunctions::print(get_openxr_api()->get_error_string(result));
+		xrDestroyEnvironmentDepthProviderMETA(provider);
+		provider = nullptr;
+		return false;
+	}
+
+	XrEnvironmentDepthSwapchainStateMETA swapchain_state = {
+		XR_TYPE_ENVIRONMENT_DEPTH_SWAPCHAIN_STATE_META, // type
+		nullptr, // next
+	};
+
+	result = xrGetEnvironmentDepthSwapchainStateMETA(swapchain, &swapchain_state);
+	if (XR_FAILED(result)) {
+		UtilityFunctions::print("xrGetEnvironmentDepthSwapchainStateMETA failed!");
+		UtilityFunctions::print(get_openxr_api()->get_error_string(result));
+		xrDestroyEnvironmentDepthSwapchainMETA(swapchain);
+		swapchain = nullptr;
+		xrDestroyEnvironmentDepthProviderMETA(provider);
+		provider = nullptr;
+		return false;
+	}
+
+	uint32_t image_count;
+	result = xrEnumerateEnvironmentDepthSwapchainImagesMETA(swapchain, 0, &image_count, nullptr);
+	if (XR_FAILED(result)) {
+		UtilityFunctions::print("xrEnumerateEnvironmentDepthSwapchainImagesMETA failed to get count!");
+		UtilityFunctions::print(get_openxr_api()->get_error_string(result));
+		xrDestroyEnvironmentDepthSwapchainMETA(swapchain);
+		swapchain = nullptr;
+		xrDestroyEnvironmentDepthProviderMETA(provider);
+		provider = nullptr;
+		return false;
+	}
+
+	struct XrSwapchainImageOpenGLKHR {
```

Comment:
This is fishy. How do I know which graphics API to use? With trial and error I have determined that it is OpenGL (not ES, as I thought), at least when running it through Quest Link.

Should there be some #ifdefs here? Or can it be checked at runtime? Though if it is Vulkan (or Direct3D, but I assume that is not supported), the struct layout is different. But I assume they can be converted to the same RIDs.